markets/perps-market/contracts/storage/PerpsAccount.sol

```diff
@@ -113,6 +119,41 @@ library PerpsAccount {
         }
     }
 
+    /**
+     * @notice This function applies the pnl of a closing position to the account
+     * @dev It will either reduce the account's debt or increase the account's debt
+     * @dev It will also update the account's collateral amount if the debt is fully paid off
+     */
+    function applyPnl(Data storage self, int256 pnl) internal {
+        if (pnl > 0) {
+            int256 leftoverDebt = self.debt.toInt() - pnl;
+            if (leftoverDebt > 0) {
+                self.debt = leftoverDebt.toUint();
+            } else {
+                self.debt = 0;
+                updateCollateralAmount(self, SNX_USD_MARKET_ID, -leftoverDebt);
+            }
+        } else {
+            int256 creditAvailable = self.collateralAmounts[SNX_USD_MARKET_ID].toInt();
+            int256 leftoverCredit = creditAvailable + pnl;
+
+            if (leftoverCredit > 0) {
+                updateCollateralAmount(self, SNX_USD_MARKET_ID, pnl);
+            } else {
+                updateCollateralAmount(self, SNX_USD_MARKET_ID, -creditAvailable);
+                self.debt += (-leftoverCredit).toUint();
+            }
+        }
+    }
+
+    function isEligibleForMarginLiquidation(
+        Data storage self,
+        PerpsPrice.Tolerance stalenessTolerance
+    ) internal view returns (bool isEligible, int256 availableMargin) {
+        availableMargin = getAvailableMargin(self, stalenessTolerance);
```

Comment:
Does this update the latest pnl numbers before giving back the number? 
I mean assume someone is elegible for liquidation, but his debt doesn't show, I think we need to update debt number somewhere?

Reply:
yes, please see `getAvailableMargin` logic.